go/updater/updater.go

```diff
@@ -307,7 +310,10 @@ func (u *Updater) apply(src string, dest string) (tmpPath string, err error) {
 
 // Update checks, downloads and performs an update.
 func (u *Updater) Update(ui libkb.UpdateUI, force bool, requested bool) (update *keybase1.Update, err error) {
-	update, err = u.CheckForUpdate(false, force, requested)
+	u.Lock()
+	defer u.Unlock()
```

Comment:
yeah misspoke when said single flight... it is only a mutex... but this is ok since anything that piles up will return immediately with the time check that occurs first thing.

In order to test this I made the ticker run once a second (and the update check for 1 minute). So there were pileups but they all returned immediately in the time check since the update time < 1 minute.